### ISSUE TITLE
Training process

### ISSUE THREAD

**Issue by ChimPansky** [Recursive-Descenders/DSLR#3]
So help me confirm whether i understand the training/predicting process correctly:

We have: 

- **dataset_train**.csv (**1600** records **including** houses)
- **dataset_test**.csv (**400** records **excluding** houses)

1. in **training**, we use the **full dataset_train.csv** for creating the **model.json**.
2. in **predicting**, we use **model.json** on the **full dataset_train.csv** to **predict the houses**, which we then can **compare** to the **houses in dataset_train.csv**. This is where we get the **accuracy** from.

did i get that right?

For **testing purposes** i did the following:

1. I **split** the **dataset_train.csv** in **2 parts** (first 800 and second 800)
2. then did the **training** on **dataset_train_first_half.csv** to create a **model_first_half.json**
3. then did **predicting** on **dataset_train_second_half.csv** with **model_first_half.json**

this is the **outcome**:

<img width="827" height="93" alt="Image" src="https://github.com/user-attachments/assets/bc1f9a6d-6898-4cbe-8620-c9aaffece76c" />

.
i also did it the **other way around** (predicting on **dataset_train_first_half** with **model_second_half**):

<img width="839" height="86" alt="Image" src="https://github.com/user-attachments/assets/ef406032-8cb0-4521-94be-1a05f731167a" />
.

as well as predicting on **dataset_train_second_half.csv**  with **model_second_half.json**:

<img width="835" height="90" alt="Image" src="https://github.com/user-attachments/assets/abcf1999-67e0-4d1e-9a0e-3f8e70b98963" />
.

Not exactly sure what to do with this, just wanted to share with you.

**Comment by zolfagharipour** [comment]
To a certain extend this is also my problem. The test data doesnt have labels (Its not given which house they belong to).
So at the moment I train on all ``dataset_train`` and predict with ``dataset_test`` and save the file as the **Subject** expects. But as you mentioned I cant find accuracy like this and this may be Problematic with **98%** accuracy which **Subject** requires for evaluation.

- I can train on subset of  ``dataset_train`` and use the rest to check my model as you did, or maybe other tricks. For that Im waiting to meet my machine learning friend soon to discuss the best Options.
- You have done more than what I expected in the right direction for last **Pull request**, Thanks. The real solution to accuracy was not part of it yet. (WIP)

**Comment by zolfagharipour** [comment]
In addition if we can get our hand on **Eval sheet** we may have better Idea of how the accuracy is being evaluated.

**Comment by ChimPansky** [comment]
> To a certain extend this is also my problem. The test data doesnt have labels (Its not given which house they belong to). So at the moment I train on all `dataset_train` and predict with `dataset_test` and save the file as the **Subject** expects. But as you mentioned I cant find accuracy like this and this may be Problematic with **98%** accuracy which **Subject** requires for evaluation.
> 
> * I can train on subset of  `dataset_train` and use the rest to check my model as you did, or maybe other tricks. For that Im waiting to meet my machine learning friend soon to discuss the best Options.
> * You have done more than what I expected in the right direction for last **Pull request**, Thanks. The real solution to accuracy was not part of it yet. (WIP)

okay, lets wait for the input of your ML friend then. 
I was just not expecting that training on second half + predicing on second half would give lower than 98%. but it could also be a data quality issue. e.g. there are more exceptions in the second half where students, who from their skillset would fit into a certain house, would actually be member of a different house (like Hermione - i mean, she for sure should be in Ravenclaw).
I guess that is a general problematic in ML: if your training data is too small, then your model is less precise

**Comment by ChimPansky** [comment]
> In addition if we can get our hand on **Eval sheet** we may have better Idea of how the accuracy is being evaluated.

i found the eval sheet:
[dslr.pdf](https://github.com/user-attachments/files/26853510/dslr.pdf)

but i guess we would need the dataset_truth.csv for the list of houses of dataset_test.csv
mb we find it later

**Comment by zolfagharipour** [comment]
- I found the ``dataset_truth.csv`` (sent you on discord)
- I will create another PR with new app which takes the ``truth`` and use **Scikit-Learn** library as **Subject** suggested for accuracy.
- Yes you are right with data inconsistency and that your dataset is important in your accuracy. In theory I could use more optimization tricks but that would be **Bonus**.
